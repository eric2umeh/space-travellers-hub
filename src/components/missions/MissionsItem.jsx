import { useSelector, useDispatch } from 'react-redux';

import { missionsActions } from '../../redux/missions/missionsSlice';
import classes from './MissionsItem.module.css';

const Missions = () => {
  const missions = useSelector((state) => state.missions.missions);
  const dispatch = useDispatch();

  let missionsContent = <p>There is no Mission content</p>;
  if (missions.length > 0) {
    missionsContent = (
      <table className={classes.table}>
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th>Actions</th>
          </tr>
        </thead>
        <tbody className={classes.table_body}>
          {missions.map((mission) => (
            <tr key={mission.mission_id}>
              <td className={classes.name}>{mission.mission_name}</td>
              <td className={classes.description}>{mission.description}</td>
              <td className={classes.status}>
                <span className={classes.not_a_member}>Not A Member</span>
              </td>
              <td className={classes.actions}>
                {missions.reserved ? (
                  <button
                    onClick={() => dispatch(missionsActions.leaveMissionAction(mission.mission_id))}
                    className={classes.btn_leavemission}
                    type="button"
                  >
                    Leave Mission
                  </button>
                ) : (
                  <button
                    onClick={() => dispatch(missionsActions.joinMissionAction(mission.mission_id))}
                    className={classes.btn_join_mission}
                    type="button"
                  >
                    Join Mission
                  </button>
                )}
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    );
  }
  return <div className={classes.container}>{missionsContent}</div>;
};

export default Missions;
