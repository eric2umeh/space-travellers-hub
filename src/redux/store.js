import { configureStore } from '@reduxjs/toolkit';
import rockets from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rockets.reducer,
  },
});

export default store;
