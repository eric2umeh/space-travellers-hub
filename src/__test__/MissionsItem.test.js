import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import MissionsItem from '../components/missions/MissionsItem';
import store from '../redux/store';

const App = () => {
  <Provider store={store}>
    <MissionsItem />
  </Provider>;
};

describe('Mission component test', () => {
  it('Testing Mission page rendering', () => {
    const MissionComponent = renderer.create(<App />).toJSON();
    expect(MissionComponent).toMatchSnapshot();
  });
});
