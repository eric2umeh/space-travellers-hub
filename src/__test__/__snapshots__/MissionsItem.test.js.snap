// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission component test Testing Mission page rendering 1`] = `null`;
